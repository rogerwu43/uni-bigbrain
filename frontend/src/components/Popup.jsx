import React from 'react';
import PropTypes from 'prop-types';
import { Modal, ModalContent, ModalBackground } from '../styles/Modal.styles';
import { BlueMediumButton } from '../styles/Global.styles'
import OutsideClickHandler from 'react-outside-click-handler';

function Popup ({ closeModal, content, active }) {
  return (
    <ModalBackground active={active}>
      <OutsideClickHandler onOutsideClick={closeModal}>
        <Modal active={active}>
          <ModalContent>
            <p>{content}</p>
            <BlueMediumButton onClick={closeModal}>OK</BlueMediumButton>
          </ModalContent>
        </Modal>
       </OutsideClickHandler>
    </ModalBackground>
  );
}
export default Popup;

Popup.propTypes = {
  content: PropTypes.string,
  active: PropTypes.bool,
  closeModal: PropTypes.func
}
