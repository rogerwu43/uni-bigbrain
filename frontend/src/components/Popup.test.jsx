import { mount } from 'enzyme'
import React from 'React'
import renderer from 'react-test-renderer'
import Popup from './Popup'

describe('Popup', () => {
  const noop = () => {};

  it('has a custom message', () => {
    const message = 'A custom message';
    const popup = mount(<Popup content={message} closeModal={noop}/>);
    expect(popup.find('p').text()).toBe(message);
  });

  it('uses the close function prop on button press', () => {
    const fn = jest.fn();
    const popup = mount(<Popup closeModal={fn}/>);
    popup.find('button').simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('is visible with when the active prop is set to true', () => {
    const popup = mount(<Popup closeModal={noop} active={true} />);
    expect(popup.getDOMNode()).toHaveStyle('display: block');
  });

  it('is not visible with when the active prop is set to false', () => {
    const popup = mount(<Popup closeModal={noop} active={false} />);
    expect(popup.getDOMNode()).toHaveStyle('display: none');
  });

  // snapshots
  it('renders with minimal props (snapshot)', () => {
    const popup = renderer.create(<Popup closeModal={noop}/>).toJSON()
    expect(popup).toMatchSnapshot()
  });

  it('renders with custom text (snapshot)', () => {
    const popup = renderer.create(<Popup content = 'A custom message' closeModal={noop}/>).toJSON()
    expect(popup).toMatchSnapshot()
  });
});
