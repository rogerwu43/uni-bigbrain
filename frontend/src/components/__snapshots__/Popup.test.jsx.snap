// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup renders with custom text (snapshot) 1`] = `
<div
  className="sc-hKwCoD gqnsJx"
>
  <div>
    <div
      className="sc-bdvvaa jcuZkM"
    >
      <div
        className="sc-gsDJrp flQMAA"
      >
        <p>
          A custom message
        </p>
        <button
          className="sc-furvIG sc-bqiQRQ kGJSNV gwxRFW"
          onClick={[Function]}
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Popup renders with minimal props (snapshot) 1`] = `
<div
  className="sc-hKwCoD gqnsJx"
>
  <div>
    <div
      className="sc-bdvvaa jcuZkM"
    >
      <div
        className="sc-gsDJrp flQMAA"
      >
        <p />
        <button
          className="sc-furvIG sc-bqiQRQ kGJSNV gwxRFW"
          onClick={[Function]}
        >
          OK
        </button>
      </div>
    </div>
  </div>
</div>
`;
